**OBLIGATORIO FINAL G448 - Data Wrangling**
<br>**INTEGRANTES:**
<br>
+ Theo Pereira, 5703834-3, theopereira2001@gmail.com
+ Carlos Píriz, 4298784-4, piriz.carlos@gmail.com
+ Gabriel	Better,3138871-2, gabrielbetter@gmail.com

In [1]:
### Importacion de librerias ###

# Sistema
import os

# Tratamiento de datos
import pandas as pd
import numpy as np
import datetime as dt
import math

# Visualizacion 
import seaborn as sns
import matplotlib as mpl
from matplotlib import ticker
from matplotlib.ticker import FuncFormatter
from babel.numbers import format_currency
import matplotlib.pyplot as plt
%matplotlib inline

**CONSIGNAS**
<br>
El objetivo general es realizar un análisis exploratorio de los anuncios inmobiliarios de
algunas provincias de la Patagonia Argentina reportados por el portal Airbnb.
<br>
Los datos están conformados por cuatro archivos csv:
+ Neuquen.csv
+ Rio_Negro.csv
+ Chubut.csv
+ Tierra_del_Fuego.csv

**<h1>1. Carga de datos y armado del dataset</h1>**

In [2]:
# Carga de archivos de datos de chubut
df_chubut = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Chubut.csv')
df_chubut["provincia"]="CHUBUT"
df_chubut.sample(10)

id_alojamiento                                               name  \
134            38028114                                              Fuego   
56   845848067167341979                                    Almacén de Amor   
245            13941382                          Cabañas MARI MARI (PUELO)   
106  919842831297441981                                 2x1 Central Family   
122  782580256541651315                    Casa familiar con cálido jardín   
23   653750937066514352                                     Trelew Apart 7   
18             53472591  On the Spot!(5pax) Pto Madryn - centro 50m del...   
255  803666373797660065                      Casa nueva equipada con patio   
167  904514761867865527                              Departamento Arcoiris   
240  894918223105948004                                  hospedaje Lorenzo   

         category     rating                city   latitude  longitude  \
134   entire_home   4,86 (7)              Trelew -43.262050 -65.302140   
56   private_room    4,5 (4)              Esquel -42.919560 -71.331530   
245   entire_home        NaN             El Hoyo -42.057100 -71.541730   
106   entire_home      Nuevo       Puerto Madryn -42.768637 -65.035316   
122  private_room    5,0 (4)              Rawson -43.295951 -65.085494   
23    entire_home    5,0 (7)              Trelew -43.256140 -65.311150   
18    entire_home  4,77 (44)       Puerto Madryn -42.767826 -65.033134   
255   entire_home    5,0 (5)       Puerto Madryn -42.787870 -65.024040   
167   entire_home      Nuevo              Trelew -43.245050 -65.298660   
240   entire_home      Nuevo  Comodoro Rivadavia -45.859679 -67.496781   

                                       characteristics    check_in  \
134  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-08-01   
56   ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-01   
245  ['No hay evaluaciones', '3 camas', '2 dormitor...  2023-07-01   
106          ['4 camas', '3 dormitorios', '2,5 baños']  2023-07-01   
122  ['1 ,5 baños', '1 cama doble', '2 camas de una...  2023-07-01   
23   ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-07-01   
18   ['3 camas', '2 dormitorios', '1 baño', '2 cama...  2023-07-01   
255  ['4 camas', '1 dormitorio', '1 baño', 'Calific...  2023-07-02   
167               ['1 cama', '1 dormitorio', '1 baño']  2023-07-01   
240              ['2 camas', '1 dormitorio', '1 baño']  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
134  2023-08-07              NaN        $48 USD           noche   
56   2023-07-06              NaN        $12 USD           noche   
245  2023-07-06              NaN        $59 USD           noche   
106  2023-07-06              NaN       $120 USD           noche   
122  2023-07-06              NaN        $36 USD           noche   
23   2023-07-06          $35 USD        $50 USD           noche   
18   2023-07-06              NaN        $61 USD           noche   
255  2023-07-07              NaN        $64 USD           noche   
167  2023-07-06              NaN        $46 USD           noche   
240  2023-07-06          $38 USD        $48 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
134             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
56              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
245             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
106             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
122             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
23              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
18              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
255             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
167             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
240             1000      2023-06-24       202

In [3]:
# Carga de archivos de datos de Neuquen
df_neuquen = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Neuquen.csv')

df_neuquen["provincia"] = "NEUQUEN"

df_neuquen.sample(10)

id_alojamiento                                             name  \
99   918988836999320780                                          Al Paso   
265            31265760               Barrio Arevalo - AMUEBLADO COMAHUE   
70   837127197406674087                         2 ambientes en el centro   
0              42980235  Depto. Centro de Neuquén. Wifi. Late check out.   
102  809747842069626522                          Casa - Quincho y parque   
14   823188407059186857  Excelente dpto moderno con cochera. zona centro   
242  839758121679938488                  Dpto en el centro de Cipolletti   
259  853518125031245252                    Alquiler Bahia Manzano Resort   
96   858615923911091711                                 Monoambiente H&O   
226  720383497670578896    Agradable casa \ncon estacionamiento privado.   

        category      rating        city   latitude  longitude  \
99   entire_home       Nuevo    Plottier -38.955203 -68.225258   
265  entire_home         NaN  Cipolletti -38.936050 -68.011990   
70   entire_home       Nuevo     Neuquén -38.949853 -68.052640   
0    entire_home  4,71 (165)     Neuquén -38.952190 -68.057380   
102  entire_home         NaN     Neuquén -38.934354 -68.123681   
14   entire_home    5,0 (18)     Neuquén -38.949830 -68.054759   
242  entire_home         NaN  Cipolletti -38.939177 -68.001222   
259  entire_home         NaN     Neuquén -38.960750 -68.098340   
96   entire_home       Nuevo  Cipolletti -38.939182 -68.001279   
226  entire_home         NaN  Cipolletti -38.938190 -67.975130   

                                       characteristics    check_in  \
99   ['2 camas', '1 dormitorio', '1 baño', 'Estacio...  2023-07-01   
265  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
70   ['No hay evaluaciones', '2 camas', '1 dormitor...  2023-07-01   
0    ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
102  ['No hay evaluaciones', '4 camas', '3 dormitor...  2023-07-03   
14   ['2 camas', '1 dormitorio', '1 ,5 baños', '1 c...  2023-08-09   
242  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-09-04   
259  ['No hay evaluaciones', '3 camas', '1 dormitor...  2023-07-01   
96       ['2 camas', '1 dormitorio', '1 baño', 'Wifi']  2023-07-01   
226  ['5 camas', '2 dormitorios', '1 ,5 baños', '1 ...  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
99   2023-07-06              NaN        $73 USD           noche   
265  2023-07-06              NaN        $52 USD           noche   
70   2023-07-06          $33 USD        $42 USD           noche   
0    2023-07-06              NaN        $23 USD           noche   
102  2023-07-08              NaN        $33 USD           noche   
14   2023-08-14              NaN        $42 USD           noche   
242  2023-09-10              NaN        $79 USD           noche   
259  2023-07-08              NaN        $90 USD           noche   
96   2023-07-06              NaN        $72 USD           noche   
226  2023-07-06              NaN        $79 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
99               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
265              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
70               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
0                298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
102              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
14               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
242              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
259              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
96               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
226              298      2023-06-24       2023-05-01  239.919355   NEUQUEN

In [4]:
# Carga de archivos de datos de Rio Negro
df_rio_negro = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Rio_Negro.csv')

df_rio_negro["provincia"] = "RIO NEGRO"

df_rio_negro.sample(10)

id_alojamiento                                               name  \
188            15965363      Villa La Angostura "Casa Aimé" - Amazing view   
101  794001040519893954                              Tillka - Cabaña Terra   
151  554319517238694179                    Cabaña Raulí para 8 con jacuzzi   
109  571693285038080905  Luminoso y vista al lago en el centro de Baril...   
192            33134546                                   L&L Studio Cabin   
40   881218792320627389                            Tiny house en el bosque   
153            46080428         Casa Náutica La Cascada con Vista al Lago.   
33             48316755                  Beautiful studio with great views   
100            37496800          Increíble vista en el centro de Bariloche   
256            12292962   Depto Vista al Lago & Isla Victoria en Llao Llao   

        category      rating                     city   latitude  longitude  \
188  entire_home    5,0 (49)                 La Villa -40.779473 -71.663992   
101  entire_home    5,0 (10)       Villa La Angostura -40.763185 -71.640431   
151  entire_home   4,94 (17)  San Carlos de Bariloche -41.145020 -71.412740   
109  entire_home   4,47 (49)  San Carlos de Bariloche -41.134450 -71.316020   
192  entire_home  4,76 (140)  San Carlos de Bariloche -41.128610 -71.370400   
40   entire_home       Nuevo  San Martin de los Andes -40.095194 -71.303187   
153  entire_home    4,8 (49)  San Carlos de Bariloche -41.119660 -71.376270   
33   entire_home    5,0 (55)  San Carlos de Bariloche -41.122440 -71.389390   
100  entire_home  4,79 (135)  San Carlos de Bariloche -41.134130 -71.310850   
256  entire_home   4,98 (44)  San Carlos de Bariloche -41.043290 -71.517720   

                                       characteristics    check_in  \
188  ['8 camas', '4 dormitorios', '1 cama king', '7...  2023-08-05   
101  ['3 camas', '1 dormitorio', '1 baño', 'Estacio...  2023-07-01   
151  ['7 camas', '4 dormitorios', '1 ,5 baños', '2 ...  2023-09-05   
109  ['2 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-06   
192  ['1 cama', 'Monoambiente', '1 baño', '1 cama d...  2023-08-04   
40   ['1 cama', '1 dormitorio', '1 baño', 'Estacion...  2023-07-01   
153  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-11   
33   ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-18   
100  ['2 baños', '1 cama doble', '2 camas de una pl...  2024-02-15   
256  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-23   

      check_out price_discounted price_original price_qualifier  \
188  2023-08-10              NaN       $450 USD           noche   
101  2023-07-06              NaN       $125 USD           noche   
151  2023-09-11         $157 USD       $185 USD           noche   
109  2023-07-11              NaN       $120 USD           noche   
192  2023-08-10              NaN        $30 USD           noche   
40   2023-07-06              NaN        $75 USD           noche   
153  2023-07-16              NaN       $168 USD           noche   
33   2023-07-23              NaN        $70 USD           noche   
100  2024-02-20              NaN        $65 USD           noche   
256  2023-07-30              NaN        $77 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
188             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
101             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
151             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
109             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
192             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
40              1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
153             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
33              1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
100             1000      2023-06-24       2023-05-01  239.9

In [5]:
# Carga de archivos de datos de Tierra delFuego
df_tierra_del_fuego = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Tierra_del_Fuego.csv')

df_tierra_del_fuego["provincia"] = "TIERRA DEL FUEGO"

df_tierra_del_fuego.sample(10)

id_alojamiento                                               name  \
117            53699852                            Amundsen Apartment PB I   
240  828090107927300389                                          GRUshuaia   
217             6031821                Habitación  Canela - 2 camas Centro   
90   570310791164532435                              Casa barrio de sueños   
55   629517965149568616      Viví la conexión con la naturaleza y cercanía   
264  568003608911978583  Hermoso departamento con vista al canal y montaña   
172  690413283545175725                                      Las Aguilas 3   
180            33710998                                    Disfrutando Sur   
104            53989333  VISTAS MARAVILLOSASDEL DEPARTAMENTO DE CHUNCHU...   
80   866795295366864263                     Alvere ll Temporary Apartments   

         category      rating     city   latitude  longitude  \
117   entire_home   4,95 (42)  Ushuaia -54.808340 -68.319750   
240   entire_home         NaN  Ushuaia -54.815088 -68.329166   
217  private_room  4,72 (274)  Ushuaia -54.806960 -68.327230   
90    entire_home   4,62 (21)  Ushuaia -54.840640 -68.376460   
55    entire_home     5,0 (9)  Ushuaia -54.794079 -68.287552   
264   entire_home         NaN  Ushuaia -54.813980 -68.337660   
172   entire_home    4,67 (9)  Ushuaia -54.814922 -68.341647   
180   entire_home   4,95 (80)  Ushuaia -54.814190 -68.325200   
104   entire_home   4,96 (55)  Ushuaia -54.809720 -68.320980   
80    entire_home         NaN  Ushuaia -54.804440 -68.325990   

                                       characteristics    check_in  \
117  ['1 cama', 'Monoambiente', '1 baño', '1 cama d...  2023-07-03   
240  ['No hay evaluaciones', '1 cama', 'Monoambient...  2023-09-13   
217  ['2 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-01   
90   ['1 cama', '1 dormitorio', '2 baños', 'Wifi', ...  2023-07-07   
55   ['2 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-01   
264  ['1 cama', 'Monoambiente', '1 baño', 'Wifi', '...  2023-08-15   
172  ['4 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-08   
180  ['2 camas', 'Monoambiente', '1 baño', '2 camas...  2023-07-11   
104  ['4 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-02   
80   ['2 camas', 'Monoambiente', '1 baño', '2 camas...  2023-07-05   

      check_out price_discounted price_original price_qualifier  \
117  2023-07-08              NaN        $55 USD           noche   
240  2023-09-18          $51 USD        $85 USD           noche   
217  2023-07-06          $33 USD        $40 USD           noche   
90   2023-07-12              NaN        $76 USD           noche   
55   2023-07-07              NaN        $55 USD           noche   
264  2023-08-20              NaN        $47 USD           noche   
172  2023-07-14              NaN        $50 USD           noche   
180  2023-07-16              NaN        $57 USD           noche   
104  2023-07-07              NaN        $85 USD           noche   
80   2023-07-10              NaN        $26 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
117             1000      2023-06-24       2023-05-01  239.919355   
240             1000      2023-06-24       2023-05-01  239.919355   
217             1000      2023-06-24       2023-05-01  239.919355   
90              1000      2023-06-24       2023-05-01  239.919355   
55              1000      2023-06-24       2023-05-01  239.919355   
264             1000      2023-06-24       2023-05-01  239.919355   
172             1000      2023-06-24       2023-05-01  239.919355   
180             1000      2023-06-24       2023-05-01  239.919355   
104             1000      2023-06-24       2023-05-01  239.919355   
80              1000      2023-06-24       2023-05-01  239.919355   

            provincia  
117  TIERRA DEL FUEGO  
240  TIERRA DEL FUEGO  
217  TIERRA DEL FUEGO  
90   TIERRA DEL FUEGO  
55   TIERRA DEL FUEGO  
264  TIERRA DEL FUEGO  
172  TIE

In [6]:
# Verificacion de columnas para concatenar
(df_chubut.columns == df_neuquen.columns) & (df_rio_negro.columns == df_tierra_del_fuego.columns) & (df_chubut.columns == df_tierra_del_fuego.columns)

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True])

In [7]:
# Concatenacion de los dataframes separados por provincias en un unico archivo
df_alquileres = pd.concat([df_chubut, df_neuquen, df_rio_negro, df_tierra_del_fuego])
df_alquileres.sample(10)

id_alojamiento                                          name  \
145  828844032499644191  Disfrutá tu estadía en una zona privilegiada   
113            48254202                                    Gartehütte   
212            53336620          Comodo Monoambiente en puerto madryn   
137            30321832           Alquiler de departamento temporario   
234            42638951                          Departamento en Loft   
52             34244276                Patagonia Apartments 2 ¡Nuevo!   
228  858196681965117698                      La Tempranera del Beagle   
215  878902407464873092                         Monoambiente en suite   
211  616671191595254070                                    En_Ushuaia   
141            39108767                     Mar y ballenas. Sarmiento   

        category      rating                     city   latitude  longitude  \
145  entire_home    4,9 (10)                  Ushuaia -54.810940 -68.331786   
113  entire_home   4,96 (50)  San Carlos de Bariloche -41.110810 -71.425520   
212  entire_home    5,0 (16)            Puerto Madryn -42.767020 -65.037330   
137  entire_home   4,43 (35)                  Neuquén -38.945640 -68.052460   
234  entire_home    4,67 (3)                  Neuquén -38.959300 -68.155430   
52   entire_home  4,96 (103)                  Ushuaia -54.813240 -68.326250   
228  entire_home         NaN                  Ushuaia -54.841288 -68.381704   
215  entire_home       Nuevo                  Neuquén -38.954110 -68.084270   
211  entire_home   4,69 (26)                  Ushuaia -54.813538 -68.325592   
141  entire_home    4,9 (39)            Puerto Madryn -42.768300 -65.032680   

                                       characteristics    check_in  \
145  ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-30   
113  ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-09-06   
212  ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-07-03   
137  ['1 cama', 'Monoambiente', '1 baño', 'Wifi', '...  2023-07-01   
234  ['2 camas', '1 dormitorio', 'Baño de visitas',...  2023-07-01   
52   ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-07-14   
228  ['No hay evaluaciones', '5 camas', '2 dormitor...  2023-07-01   
215  ['2 camas', 'Monoambiente', '1 baño', 'Estacio...  2023-07-01   
211  ['3 camas', '2 dormitorios', '1 baño', '1 cama...  2023-07-01   
141  ['5 camas', '2 dormitorios', '2 baños', 'Calif...  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
145  2023-08-05              NaN        $34 USD           noche   
113  2023-09-12              NaN        $30 USD           noche   
212  2023-07-10          $48 USD        $55 USD           noche   
137  2023-07-06              NaN        $34 USD           noche   
234  2023-07-06              NaN        $77 USD           noche   
52   2023-07-19              NaN        $57 USD           noche   
228  2023-07-06         $133 USD       $166 USD           noche   
215  2023-07-06          $53 USD        $66 USD           noche   
211  2023-07-07          $49 USD        $68 USD           noche   
141  2023-07-06          $80 USD       $140 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
145             1000      2023-06-24       2023-05-01  239.919355   
113             1000      2023-06-24       2023-05-01  239.919355   
212             1000      2023-06-24       2023-05-01  239.919355   
137              298      2023-06-24       2023-05-01  239.919355   
234              298      2023-06-24       2023-05-01  239.919355   
52              1000      2023-06-24       2023-05-01  239.919355   
228             1000      2023-06-24       2023-05-01  239.919355   
215              298      2023-06-24       2023-05-01  239.919355   
211             1000      2023-06-24       2023-05-01  239.919355   
141             1000      2023-06-24       2023-05-01  239.919355   

            provincia  
145  TIERRA DEL FUEGO  
113         RIO NE

In [8]:
# Reseteo del index, permitiendo duplicados porque aun no sabemos si son unicos los valores
# df_alquileres.reset_index(names = 'id_alojamiento', drop=True,inplace=True,allow_duplicates= True)
# df_alquileres.reset_index(names = 'id_alojamiento', drop=True,inplace=True,allow_duplicates= False)
df_alquileres.reset_index(drop=True,inplace=True)

*Verificacion de filas de los dataframes*

In [9]:
# Verificacion de la cantidad de filas de cada dataframe
print("La cantidad de filas del dataframe de Chubut es:"+str(df_chubut.shape[0]))
print("La cantidad de filas del dataframe de Neuquen es:"+str(df_neuquen.shape[0]))
print("La cantidad de filas del dataframe de Rio Negro es:"+str(df_rio_negro.shape[0]))
print("La cantidad de filas del dataframe de Tierra del fuego es:" + str(df_tierra_del_fuego.shape[0]))
print("La cantidad de filas del dataframe unificado de alquileres es:" + str(df_alquileres.shape[0]))


La cantidad de filas del dataframe de Chubut es:270
La cantidad de filas del dataframe de Neuquen es:270
La cantidad de filas del dataframe de Rio Negro es:270
La cantidad de filas del dataframe de Tierra del fuego es:270
La cantidad de filas del dataframe unificado de alquileres es:1080


*Verificacion de las columnas de los dataframes*

In [10]:
# Verificacion de la cantidad de columnas de cada dataframe
print("La cantidad de columnas del dataframe de Chubut es:" +
      str(df_chubut.shape[1]))
print("La cantidad de columnas del dataframe de Neuquen es:" +
      str(df_neuquen.shape[1]))
print("La cantidad de columnas del dataframe de Rio Negro es:" +
      str(df_rio_negro.shape[1]))
print("La cantidad de columnas del dataframe de Tierra del fuego es:" +
      str(df_tierra_del_fuego.shape[1]))
print("La cantidad de columnas del dataframe unificado de alquileres es:" +
      str(df_alquileres.shape[1]))

La cantidad de columnas del dataframe de Chubut es:18
La cantidad de columnas del dataframe de Neuquen es:18
La cantidad de columnas del dataframe de Rio Negro es:18
La cantidad de columnas del dataframe de Tierra del fuego es:18
La cantidad de columnas del dataframe unificado de alquileres es:18


*Verificacion general e inicial del dataframe unificado*

In [11]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1080 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_alojamiento    1080 non-null   int64  
 1   name              1080 non-null   object 
 2   category          1080 non-null   object 
 3   rating            886 non-null    object 
 4   city              1076 non-null   object 
 5   latitude          1080 non-null   float64
 6   longitude         1080 non-null   float64
 7   characteristics   1080 non-null   object 
 8   check_in          1080 non-null   object 
 9   check_out         1080 non-null   object 
 10  price_discounted  201 non-null    object 
 11  price_original    1080 non-null   object 
 12  price_qualifier   1080 non-null   object 
 13  cant_resultados   1080 non-null   int64  
 14  fecha_scrapping   1080 non-null   object 
 15  fecha_cambio_bna  1080 non-null   object 
 16  cambio_bna        1080 non-null   float64


In [12]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.describe()

id_alojamiento     latitude    longitude  cant_resultados   cambio_bna
count    1.080000e+03  1080.000000  1080.000000      1080.000000  1080.000000
mean     3.685341e+17   -44.426201   -68.723924       824.500000   239.919355
std      3.903051e+17     6.164758     2.159715       304.115744     0.000000
min      4.468310e+05   -54.844914   -71.688000       298.000000   239.919355
25%      3.529220e+07   -47.854503   -71.310280       824.500000   239.919355
50%      5.408344e+07   -42.003780   -68.305610      1000.000000   239.919355
75%      7.872495e+17   -39.609811   -68.049328      1000.000000   239.919355
max      9.198428e+17   -38.874370   -62.920960      1000.000000   239.919355

In [13]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.duplicated(keep="first").sum()

51

In [14]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.isna().sum()

id_alojamiento        0
name                  0
category              0
rating              194
city                  4
latitude              0
longitude             0
characteristics       0
check_in              0
check_out             0
price_discounted    879
price_original        0
price_qualifier       0
cant_resultados       0
fecha_scrapping       0
fecha_cambio_bna      0
cambio_bna            0
provincia             0
dtype: int64

*Conclusiones iniciales*
<br>
+ El dataframe unificado, coincide en cantidad de filas y columnas correspondientes a los dataframes individuales
+ Debemos de corregir algunos tipos de datos de columnas
+ Tenemos algunas columnas para reescalar
+ Tenemos filas duplicadas para revisar
+ Tenemos valores nulos para revisar, en principio para las columnas rating y price_discounted

**<h1>2. Análisis de duplicados</h1>**

*Analisis y tratamiento de duplicados global*

In [15]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres.duplicated(keep="first").sum()

51

In [16]:
#  Listo los valores duplicados
df_alquileres[df_alquileres.duplicated(keep="first")]

id_alojamiento                                               name  \
54              40755147  La Amarilla Casa de Huéspedes (con cocina) Rut...   
72    586216477715336949  Encantador dpto. con vista al mar en Puerto Ma...   
90              37836757                                     Rincón del Sol   
113   736253141994173324           Refugio Ciprés con tina de agua caliente   
127             21925703      Hermoso Duplex con Patio y Parrilla. (4 pax).   
128   644043911246503372     Los Molles, casa familiar  en entorno natural.   
129   883174783781736766                             Espacio 2 - Rada Tilly   
138             53298317                       Mar de Olivillos Planta alta   
162             31081731                 Domos Puelo, para dos, con jacuzzi   
182             50704005   Hostel patagónico con vista 360 de la Cordillera   
189   800916516894617209  Cómodas Habitaciones, 50 m del Mar con Desayun...   
216             53336620               Comodo Monoambiente en puerto madryn   
221             54362650     "Ty Bychan" \nLa Casita. Entre raíces galesas.   
234   807474929736554545                          Cabaña de Playa. Bajada 7   
247             41936762                                       El Tornado B   
253   909336244153347770                                      Bella Vista 2   
255   803666373797660065                      Casa nueva equipada con patio   
265   914317237704080953                    Dep,AlquilerTemporario,Acogedor   
630             25286446                                    Casita Hillside   
720              4635954  ESTUDIO A 20 METROS DEL CIVICO CENTER, VISTA A...   
756             38777312                                  Cabañas Waldesruh   
757             37245663  Preciosa cabaña en el bosque de Llaollao con v...   
828   604911298357239989                        IBF SKY - Ushuaia Apartment   
832   711705563371014313                 C) Cama en cuarto mixto compartido   
846             39446425                  Cozy Flat, with incredible views.   
850             54314756                       Infinity View Pent House III   
852   604911298357239989                        IBF SKY - Ushuaia Apartment   
882   710167445018555343        Apartamento Amundsen 6J - Estudio - Ushuaia   
885             40366875                 H) Cama en cuarto mixto compartido   
893   755135392594127489  Encantadoras vistas del canal de beagle - Amun...   
922             38919535       Moderno y luminoso estudio en  pleno centro.   
943   570310791164532435                              Casa barrio de sueños   
954              2942774        Room with incredible sea and mountains view   
971             24852036                                 Mi Casa Es Su Casa   
989             53797805            Humilde hogar, en el corazón de Ushuaia   
993             48043650                                Latitud Sur Ushuaia   
1001  776661041221694371                   Lindo estudio céntrico y moderno   
1008  751515304138637905          Agradable cabaña con amplias dimensiones.   
1009  837061192294336671                             Loft del Fin del Mundo   
1010  847640105410806361                                          Mini casa   
1011            54301035  Departamento de 2 ambientes -vista al Canal Be...   
1017  831611894265914063                                       Ushuaia home   
1018            54258723                NEW APARTMENT - LAS CUMBRES USHUAIA   
1032            53445794      Departamento con excelente ubicación y vistas   
1046  626675407573680894                            Precioso Departamento A   
1054  906355891628807276                                        Grand House   
1060  858196681965117698                           La Tempranera del Beagle   
1062  562143078127144674                                           Turquesa   
1063            51759643     Departamento con vista abierta  a las montañas   
1064             8048378               DESAPARTAMENTO SAN MA

In [17]:
# Elimino los duplicados, que coinciden en toda la fila
df_alquileres.drop_duplicates(keep="first", inplace=True)
df_alquileres.shape

(1029, 18)

*Analisis y tratamiento de duplicados por columna*

+ Columna 1: id_alojamiento

In [18]:
# Busco si hay duplicados
df_alquileres['id_alojamiento'].duplicated(keep='first').sum()

1

In [19]:
# Identifico filas con esos valores duplicados
df_alquileres[df_alquileres['id_alojamiento'].duplicated(keep='first')]

id_alojamiento          name     category    rating  \
648        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
648  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
648  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
648  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
648             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

In [20]:
# Muestro las filas que corresponden a los valores duplicados, todas
df_alquileres[df_alquileres['id_alojamiento'] == 53729814]

id_alojamiento          name     category    rating  \
647        53729814  Pequeña casa  entire_home  5,0 (36)   
648        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
647  San Carlos de Bariloche -41.14768  -71.31827   
648  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
647  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   
648  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
647  2023-08-23              NaN        $39 USD           noche   
648  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
647             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
648             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

-> Analisis de los valores de las filas duplicadas.

In [21]:
#Verificamos los valores 
df_alquileres.loc[647]

id_alojamiento                                               53729814
name                                                     Pequeña casa
category                                                  entire_home
rating                                                       5,0 (36)
city                                          San Carlos de Bariloche
latitude                                                    -41.14768
longitude                                                   -71.31827
characteristics     ['1 cama', '1 dormitorio', '1 baño', '1 cama d...
check_in                                                   2023-08-18
check_out                                                  2023-08-23
price_discounted                                                  NaN
price_original                                                $39 USD
price_qualifier                                                 noche
cant_resultados                                                  1000
fecha_scrapping     

In [22]:
# Verificamos los valores
df_alquileres.loc[648]

id_alojamiento                                               53729814
name                                                     Pequeña casa
category                                                  entire_home
rating                                                       5,0 (36)
city                                          San Carlos de Bariloche
latitude                                                    -41.14768
longitude                                                   -71.31827
characteristics     ['1 cama', '1 dormitorio', '1 baño', '1 cama d...
check_in                                                   2023-08-18
check_out                                                  2023-08-23
price_discounted                                                  NaN
price_original                                                $39 USD
price_qualifier                                                 noche
cant_resultados                                                  1000
fecha_scrapping     

In [23]:
# Verificamos los valores para la variable que tendria fiferencia
df_alquileres.loc[647]["characteristics"]

"['1 cama', '1 dormitorio', '1 baño', '1 cama doble', 'Estacionamiento gratis', 'Wifi']"

In [24]:
# Verificamos los valores para la variable que tendria fiferencia
df_alquileres.loc[648]["characteristics"]

"['1 cama', '1 dormitorio', '1 baño', '1 cama doble', 'Estacionamiento gratis', 'Wifi', 'Calificado con 5 de 5 en limpieza']"

*ACCION:*
<BR>
*Vemos que seria el mismo registro con excepcion de la serie characteristics, donde tiene agregada comentario de calificacion*
<br>
*Como consideramos que cada valor de id_alojamiento, deberia ser unico, eliminamos el que consideramos repetido y menos se adecua a los valores esperados por serie*

In [25]:
# Descartamos el registro que no tiene agregado el comentario de la calificacion
df_alquileres.drop([648], inplace=True)
df_alquileres[df_alquileres['id_alojamiento'] == 53729814]

id_alojamiento          name     category    rating  \
647        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
647  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
647  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
647  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
647             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

+ Columna 2: name

In [26]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['name'].duplicated(keep="first").sum()

1

In [27]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres[df_alquileres['name'].duplicated(keep="first")]

id_alojamiento                                         name  \
794  800041160997364459  Vista al Lago en el Bosque - Hab Compartida   

        category   rating                     city  latitude  longitude  \
794  shared_room  5,0 (8)  San Carlos de Bariloche  -41.1333  -71.36914   

                                       characteristics    check_in  \
794  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-05   

      check_out price_discounted price_original price_qualifier  \
794  2023-07-10              NaN        $23 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
794             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

In [28]:
df_alquileres[df_alquileres['name'].str.contains('Vista al Lago en el Bosque - Hab Compartida')]

id_alojamiento                                         name  \
712  797077259992479162  Vista al Lago en el Bosque - Hab Compartida   
794  800041160997364459  Vista al Lago en el Bosque - Hab Compartida   

        category   rating                     city  latitude  longitude  \
712  shared_room  5,0 (8)  San Carlos de Bariloche -41.13455  -71.36807   
794  shared_room  5,0 (8)  San Carlos de Bariloche -41.13330  -71.36914   

                                       characteristics    check_in  \
712  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-11   
794  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-05   

      check_out price_discounted price_original price_qualifier  \
712  2023-07-18              NaN        $21 USD           noche   
794  2023-07-10              NaN        $23 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
712             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
794             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

*ACCION:*
<br>
*En este caso vemos que los nombre coinciden pero, pueden corresponder a diferentes lugares, de hecho, estan sobre diferentes calles, segun sus coordenadas.*
<br>
*Por tal motivo los dejamos a ambos registros*

+ Columna 3: category

In [29]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['category'].duplicated(keep="first").sum()

1025

In [30]:
# Listo los duplicados segun category
df_alquileres[df_alquileres['category'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
603  760003693579176617  Acogedora Cabaña B. Residencial con Vista al C...   
641  794001040519893954                              Tillka - Cabaña Terra   
507            45275721   Casa tipo duplex muy cómoda con patio y parrilla   
28   890430561594058377                    Amplia casa con estacionamiento   
965            20441841  Acogedora habitación para 4 personas, entrada ...   
923            24852036                                 Mi Casa Es Su Casa   
76             14926605                              DAMCHE   Vista al Mar   
998  751515304138637905          Agradable cabaña con amplias dimensiones.   
41   787139916703144070                   Luz, confort y garaje, 2 del mar   
285            49819914                   Las Violetas 2 - Casa con vistas   

         category      rating                city   latitude  longitude  \
603   entire_home     5,0 (8)  Villa La Angostura -40.747480 -71.660860   
641   entire_home    5,0 (10)  Villa La Angostura -40.763185 -71.640431   
507   entire_home         NaN             Neuquén -38.943420 -68.112970   
28    entire_home       Nuevo              Esquel -42.914878 -71.334675   
965  private_room   4,82 (73)             Ushuaia -54.796460 -68.313900   
923   entire_home    4,2 (20)          Rio Grande -53.783470 -67.702090   
76    entire_home  4,76 (134)       Puerto Madryn -42.766790 -65.032010   
998   entire_home         NaN             Tolhuin -54.507803 -67.201997   
41    entire_home    5,0 (10)       Puerto Madryn -42.767240 -65.037240   
285   entire_home   4,85 (26)             Neuquén -38.943110 -68.052640   

                                       characteristics    check_in  \
603  ['4 camas', '2 dormitorios', '1 baño', '1 cama...  2023-07-03   
641  ['3 camas', '1 dormitorio', '1 baño', 'Estacio...  2023-07-01   
507  ['4 camas', '2 dormitorios', '2 baños', '4 cam...  2023-07-28   
28               ['2 camas', '1 dormitorio', '1 baño']  2023-07-01   
965  ['4 camas', '2 dormitorios', '1 baño privado',...  2023-07-13   
923  ['4 camas', '2 dormitorios', '1 baño', '4 cama...  2023-07-01   
76   ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-10-31   
998  ['8 camas', '3 dormitorios', '1 baño', 'Wifi',...  2023-07-01   
41   ['1 cama', 'Monoambiente', '1 baño', 'Califica...  2023-07-01   
285  ['2 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-02   

      check_out price_discounted price_original price_qualifier  \
603  2023-07-10          $68 USD        $75 USD           noche   
641  2023-07-06              NaN       $125 USD           noche   
507  2023-08-02              NaN        $62 USD           noche   
28   2023-07-06              NaN        $25 USD           noche   
965  2023-07-20              NaN        $21 USD           noche   
923  2023-07-06              NaN        $28 USD           noche   
76   2023-11-05              NaN        $41 USD           noche   
998  2023-07-06              NaN        $79 USD           noche   
41   2023-07-06              NaN        $45 USD           noche   
285  2023-07-08              NaN        $45 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
603             1000      2023-06-24       2023-05-01  239.919355   
641             1000      2023-06-24       2023-05-01  239.919355   
507              298      2023-06-24       2023-05-01  239.919355   
28              1000      2023-06-24       2023-05-01  239.919355   
965             1000      2023-06-24       2023-05-01  239.919355   
923             1000      2023-06-24       2023-05-01  239.919355   
76              1000      2023-06-24       2023-05-01  239.919355   
998             1000      2023-06-24       2023-05-01  239.919355   
41              1000      2023-06-24       2023-05-01  239.919355   
285              298      2023-06-24       2023-05-01  239.919355   

            provincia  
603         RIO NEGRO  
641    

*ACCION:*
<br>
*Entendemos que los valores de categoria se pueden repetir para los diferentes alquileres*
<br>
*por lo cual, no realizamos ninguna accion en base a esta columna*

+ Columna 4: rating

In [31]:
# Verifico los duplicado
df_alquileres['rating'].duplicated(keep="first").sum()

564

In [32]:
# Listamos algunas filas, repetidas por rating
df_alquileres[df_alquileres['rating'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
822            53848130                              Ocean & Mountain View   
716  782060930905714812                   Cabaña en la montaña: La Cascada   
750            45953662       Luz, montaña y naturaleza. Casa encantadora.   
333            34683504                           Loft Limay Bed&Breakfast   
603  760003693579176617  Acogedora Cabaña B. Residencial con Vista al C...   
1              51484364     Hermoso Monoambiente con Vista al Mar y Pileta   
890  866795295366864263                     Alvere ll Temporary Apartments   
684  609427764930669698                      Hermoso monoambiente céntrico   
718  602017890042571520    Suite con cocina e increíble vista a los cerros   
449  588559931833580141  Hermoso departamento en pleno centro de Cipoll...   

        category     rating                     city   latitude  longitude  \
822  entire_home   4,89 (9)                  Ushuaia -54.807163 -68.318520   
716  entire_home        NaN  San Martin de los Andes -40.144050 -71.275120   
750  entire_home    5,0 (8)  San Carlos de Bariloche -41.153810 -71.415220   
333  entire_home    5,0 (5)                  Neuquén -38.978990 -68.061380   
603  entire_home    5,0 (8)       Villa La Angostura -40.747480 -71.660860   
1    entire_home        NaN            Puerto Madryn -42.755775 -65.037265   
890  entire_home        NaN                  Ushuaia -54.804440 -68.325990   
684  entire_home  4,97 (32)  San Carlos de Bariloche -41.138149 -71.305573   
718  entire_home   5,0 (30)                 La Villa -40.774208 -71.651749   
449  entire_home        NaN               Cipolletti -38.937680 -67.990200   

                                       characteristics    check_in  \
822  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
716  ['1 cama', '1 dormitorio', '1 baño', 'Estacion...  2023-07-01   
750  ['4 camas', '2 dormitorios', '1 baño', '1 cama...  2023-08-09   
333  ['3 camas', 'Monoambiente', '1 baño', 'Estacio...  2023-07-01   
603  ['4 camas', '2 dormitorios', '1 baño', '1 cama...  2023-07-03   
1    ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-21   
890  ['2 camas', 'Monoambiente', '1 baño', '2 camas...  2023-07-05   
684  ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-09-25   
718  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
449  ['No hay evaluaciones', '2 camas', '2 dormitor...  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
822  2023-07-06          $88 USD        $98 USD           noche   
716  2023-07-06          $45 USD        $75 USD           noche   
750  2023-08-15              NaN        $63 USD           noche   
333  2023-07-06              NaN       $114 USD           noche   
603  2023-07-10          $68 USD        $75 USD           noche   
1    2023-07-26          $18 USD        $24 USD           noche   
890  2023-07-10              NaN        $26 USD           noche   
684  2023-09-30              NaN        $35 USD           noche   
718  2023-07-06              NaN        $70 USD           noche   
449  2023-07-06              NaN        $79 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
822             1000      2023-06-24       2023-05-01  239.919355   
716             1000      2023-06-24       2023-05-01  239.919355   
750             1000      2023-06-24       2023-05-01  239.919355   
333              298      2023-06-24       2023-05-01  239.919355   
603             1000      2023-06-24       2023-05-01  239.919355   
1               1000      2023-06-24       2023-05-01  239.919355   
890             1000      2023-06-24       2023-05-01  239.919355   
684             1000      2023-06-24       2023-05-01  239.919355   
718             1000      2023-06-24       2023-05-01  239.919355   
449              298      2023-06-24       2023-05-01  239.919355   

            provincia 

In [33]:
df_alquileres['rating'].unique()

array([nan, '5,0 (6)', '4,91 (70)', 'Nuevo', '4,96 (95)', '4,86 (7)',
       '4,82 (17)', '4,96 (75)', '4,91 (67)', '4,57 (14)', '4,6 (5)',
       '4,94 (18)', '4,95 (22)', '4,92 (64)', '4,94 (109)', '4,77 (44)',
       '4,89 (35)', '5,0 (7)', '4,66 (53)', '5,0 (34)', '4,98 (93)',
       '4,96 (51)', '4,82 (66)', '4,8 (51)', '4,83 (36)', '4,9 (20)',
       '5,0 (17)', '4,65 (49)', '4,7 (33)', '4,89 (9)', '4,97 (90)',
       '5,0 (10)', '4,79 (30)', '4,86 (74)', '4,86 (44)', '4,63 (128)',
       '5,0 (3)', '4,88 (16)', '4,81 (32)', '5,0 (9)', '4,93 (27)',
       '4,92 (13)', '4,5 (4)', '4,77 (13)', '4,67 (12)', '4,9 (176)',
       '4,95 (19)', '4,8 (15)', '5,0 (14)', '4,79 (14)', '5,0 (4)',
       '5,0 (19)', '4,67 (6)', '4,82 (137)', '5,0 (11)', '4,77 (102)',
       '4,47 (55)', '4,76 (134)', '4,74 (19)', '4,96 (27)', '4,91 (11)',
       '4,93 (15)', '4,62 (13)', '4,55 (33)', '4,4 (5)', '4,72 (25)',
       '4,58 (76)', '4,73 (11)', '4,83 (60)', '4,93 (28)', '4,7 (10)',
       '4,76 (85

*ACCION:*
<br>
*Entendemos que los valores de rating se pueden repetir para los diferentes alquileres*
<br>
*Por lo cual, dejamos los registros retpetidos segun esta columna*

+ Columna 5: city

In [34]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['city'].duplicated(keep="first").sum()

947

In [35]:
# Listamos repetidos, por city
df_alquileres[df_alquileres['city'].duplicated(keep="first")]

id_alojamiento                                               name  \
1               51484364     Hermoso Monoambiente con Vista al Mar y Pileta   
2     800749301724167838                           Departamento Bella Vista   
3               50769917  New Lil’ & Cozy - 100mts de la playa Puerto Ma...   
4     916738743930751566                                        Front Beach   
5               27877802      Luxury top 11th floor with the best 360 views   
...                  ...                                                ...   
1075  771725397497773450                            Casa Fueguina en Centro   
1076            26215568                               Hospedaje UshuaiAndo   
1077            42486711       Yeska C Dpto. Supercéntrico. Casco Histórico   
1078            51921207                                             KREN I   
1079  842195865716596989                                   Bahía de Ushuaia   

         category     rating           city   latitude  longitude  \
1     entire_home        NaN  Puerto Madryn -42.755775 -65.037265   
2     entire_home    5,0 (6)  Puerto Madryn -42.768007 -65.033108   
3     entire_home  4,91 (70)  Puerto Madryn -42.777590 -65.025890   
4     entire_home      Nuevo  Puerto Madryn -42.770540 -65.030810   
5     entire_home  4,96 (95)  Puerto Madryn -42.768330 -65.032100   
...           ...        ...            ...        ...        ...   
1075  entire_home  4,53 (17)        Ushuaia -54.803310 -68.302580   
1076  entire_home  4,82 (50)        Ushuaia -54.800790 -68.304230   
1077  entire_home   4,88 (8)        Ushuaia -54.805430 -68.308710   
1078  entire_home  4,85 (26)        Ushuaia -54.806170 -68.317800   
1079  entire_home        NaN        Ushuaia -54.793450 -68.276470   

                                        characteristics    check_in  \
1     ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-21   
2     ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-08-01   
3     ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-09   
4                 ['2 camas', 'Monoambiente', '1 baño']  2023-07-01   
5     ['4 camas', '2 dormitorios', '2,5 baños', '1 c...  2023-07-06   
...                                                 ...         ...   
1075  ['3 camas', '2 dormitorios', '1 baño', 'Wifi',...  2023-08-14   
1076  ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-23   
1077  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
1078  ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-10   
1079  ['1 cama', '1 dormitorio', '1 baño', 'Wifi', '...  2023-08-05   

       check_out price_discounted price_original price_qualifier  \
1     2023-07-26          $18 USD        $24 USD           noche   
2     2023-08-08              NaN        $43 USD           noche   
3     2023-07-14              NaN        $41 USD           noche   
4     2023-07-06              NaN        $35 USD           noche   
5     2023-07-11              NaN       $122 USD           noche   
...          ...              ...            ...             ...   
1075  2023-08-19          $60 USD        $76 USD           noche   
1076  2023-07-28          $45 USD        $50 USD           noche   
1077  2023-07-06              NaN        $63 USD           noche   
1078  2023-07-15              NaN        $55 USD           noche   
1079  2023-08-10              NaN        $49 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
1                1000      2023-06-24       2023-05-01  239.919355   
2                1000      2023-06-24       2023-05-01  239.919355   
3                1000      2023-06-24       2023-05-01  239.919355   
4                1000      2023-06-24       2023-05-01  239.919355   
5                1000      2023-06-24       2023-05-01  239.919355   
...               ...             ...              ...         ...   
1075             1000      2023-06-24       2023-05-01  239.919355   
10

*ACCION:*
<br>
*Entendemos que la columna city, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 6: latitude

In [36]:
# Verificamos los duplicados
df_alquileres['latitude'].duplicated(keep="first").sum()

36

In [37]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['latitude'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
1031  711709136406452484                 F) Cama en cuarto mixto compartido   
364   810083074082367782                     Departamento luminoso centrico   
967             35647657  Ushuaia. Apartamento/Vista Exclusiva. Zona cén...   
1019            53445794      Departamento con excelente ubicación y vistas   
318             34539785                    Dpto  B° Lejos de Buenos Aires.   
791              4469246  Habitación en Casita de madera y adobe. c/desa...   
1071  700712866756917764                            D) Cuarto doble privado   
296   661388148387143050  Apart céntrico 2 dorm  c/ balcon  cochera opci...   
981   742698340271989675       Excelente departamento, excelente ubicación.   
867             40366875                 H) Cama en cuarto mixto compartido   

          category      rating                     city   latitude  longitude  \
1031   shared_room    4,67 (9)                  Ushuaia -54.796494 -68.298038   
364    entire_home         NaN                  Neuquén -38.949560 -68.054970   
967    entire_home  4,95 (111)                  Ushuaia -54.805860 -68.317640   
1019   entire_home   4,91 (22)                  Ushuaia -54.808500 -68.320390   
318    entire_home   4,91 (69)                  Neuquén -38.951410 -68.114000   
791   private_room  4,68 (114)  San Carlos de Bariloche -41.073160 -71.488690   
1071  private_room   4,85 (13)                  Ushuaia -54.796494 -68.298038   
296    entire_home     5,0 (6)                  Neuquén -38.951410 -68.061740   
981    entire_home   4,94 (18)                  Ushuaia -54.810100 -68.318770   
867    shared_room   4,91 (23)                  Ushuaia -54.796459 -68.297798   

                                        characteristics    check_in  \
1031  ['1 cama', '1 dormitorio', '1,5 baños comparti...  2023-07-01   
364   ['No hay evaluaciones', '1 cama', 'Monoambient...  2023-10-02   
967   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-27   
1019  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-06   
318   ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
791   ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-09   
1071  ['2 camas', '1 dormitorio', '1,5 baños compart...  2023-07-01   
296   ['3 camas', '2 dormitorios', '1 ,5 baños', 'Wi...  2023-07-01   
981   ['3 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-08-01   
867   ['1 cama', '1 dormitorio', '1,5 baños comparti...  2023-07-01   

       check_out price_discounted price_original price_qualifier  \
1031  2023-07-06              NaN        $12 USD           noche   
364   2023-10-07              NaN        $22 USD           noche   
967   2023-08-01              NaN        $90 USD           noche   
1019  2023-07-12              NaN        $75 USD           noche   
318   2023-07-06              NaN        $47 USD           noche   
791   2023-07-14              NaN        $40 USD           noche   
1071  2023-07-06              NaN        $26 USD           noche   
296   2023-07-08              NaN        $50 USD           noche   
981   2023-08-06              NaN        $80 USD           noche   
867   2023-07-06              NaN        $12 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
1031             1000      2023-06-24       2023-05-01  239.919355   
364               298      2023-06-24       2023-05-01  239.919355   
967              1000      2023-06-24       2023-05-01  239.919355   
1019             1000      2023-06-24       2023-05-01  239.919355   
318               298      2023-06-24       2023-05-01  239.919355   
791              1000      2023-06-24       2023-05-01  239.919355   
1071             1000      2023-06-24       2023-05-01  239.919355   
296               298      2023-06-24       2023-05-01  239.919355   
981              1000      2023-06-24       2023-05-01  239.919355   
867              

In [38]:
# Vemos un ejemplo
df_alquileres[df_alquileres['latitude'] == -54.803600]

id_alojamiento                                               name  \
934             31825003  Departamento con inmejorable ubicación en Ushuaia   
1025  623178032761843625            departamento céntrico, cálido y cómodo.   

         category     rating     city  latitude  longitude  \
934   entire_home  4,9 (108)  Ushuaia  -54.8036  -68.30374   
1025  entire_home  4,88 (16)      AIU  -54.8036  -68.32132   

                                        characteristics    check_in  \
934   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
1025  ['3 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-09   

       check_out price_discounted price_original price_qualifier  \
934   2023-07-06              NaN        $85 USD           noche   
1025  2023-07-14              NaN        $33 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
934              1000      2023-06-24       2023-05-01  239.919355   
1025             1000      2023-06-24       2023-05-01  239.919355   

             provincia  
934   TIERRA DEL FUEGO  
1025  TIERRA DEL FUEGO

*ACCION:*
<br>
*Entendemos que la columna latitude, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 7: longitude

In [39]:
# Verificamos los duplicados
df_alquileres['longitude'].duplicated(keep="first").sum()

20

In [40]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['longitude'].duplicated(keep="first")].sample(10)

id_alojamiento                                              name  \
86              51311028  Amplio Monoambiente con estacionamiento,cerrado.   
1071  700712866756917764                           D) Cuarto doble privado   
15    767503846951993933                      Brisa de Mar - Puerto Madryn   
531   833878451226342434                     El lugar de Ana - Am. Comahue   
905   579038522149900540         Departamento con hermosa vista a la Bahía   
243   909336244153347770                                     Bella Vista 2   
109   905128221670422939                  Habitación con Altillo para 4 Pa   
1031  711709136406452484                F) Cama en cuarto mixto compartido   
534             41228116            Mono tipo LOFT ALTO COMAHUE SHOPPING!!   
948   889733753812060778                    Ushuaia Mágica - Ush AustraLis   

          category     rating           city   latitude  longitude  \
86     entire_home  4,55 (33)         Esquel -42.915770 -71.307150   
1071  private_room  4,85 (13)        Ushuaia -54.796494 -68.298038   
15     entire_home  4,95 (22)  Puerto Madryn -42.771634 -65.029071   
531    entire_home        NaN        Neuquén -38.953065 -68.065207   
905    entire_home  4,81 (21)        Ushuaia -54.807163 -68.318520   
243    entire_home      Nuevo         Esquel -42.917491 -71.294685   
109   private_room      Nuevo  Puerto Madryn -42.755600 -65.037001   
1031   shared_room   4,67 (9)        Ushuaia -54.796494 -68.298038   
534    entire_home        NaN        Neuquén -38.943410 -68.064100   
948    entire_home      Nuevo        Ushuaia -54.811890 -68.326950   

                                        characteristics    check_in  \
86    ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
1071  ['2 camas', '1 dormitorio', '1,5 baños compart...  2023-07-01   
15    ['3 camas', '2 dormitorios', '2,5 baños', 'Cal...  2023-08-07   
531   ['No hay evaluaciones', '3 camas', '1 dormitor...  2023-07-02   
905   ['3 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-03   
243               ['4 camas', '1 dormitorio', '1 baño']  2023-07-01   
109    ['3 camas', '1 dormitorio', '1 baño compartido']  2023-07-01   
1031  ['1 cama', '1 dormitorio', '1,5 baños comparti...  2023-07-01   
534   ['No hay evaluaciones', '2 camas', '1 dormitor...  2023-07-01   
948   ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-09   

       check_out price_discounted price_original price_qualifier  \
86    2023-07-06              NaN        $20 USD           noche   
1071  2023-07-06              NaN        $26 USD           noche   
15    2023-08-12          $34 USD        $48 USD           noche   
531   2023-07-08              NaN        $59 USD           noche   
905   2023-07-08              NaN        $77 USD           noche   
243   2023-07-06          $52 USD        $65 USD           noche   
109   2023-07-06          $14 USD        $17 USD           noche   
1031  2023-07-06              NaN        $12 USD           noche   
534   2023-07-06              NaN        $30 USD           noche   
948   2023-07-14              NaN        $45 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
86               1000      2023-06-24       2023-05-01  239.919355   
1071             1000      2023-06-24       2023-05-01  239.919355   
15               1000      2023-06-24       2023-05-01  239.919355   
531               298      2023-06-24       2023-05-01  239.919355   
905              1000      2023-06-24       2023-05-01  239.919355   
243              1000      2023-06-24       2023-05-01  239.919355   
109              1000      2023-06-24       2023-05-01  239.919355   
1031             1000      2023-06-24       2023-05-01  239.919355   
534               298      2023-06-24       2023-05-01  239.919355   
948              1000      2023-06-24       2023-05-01  239.919355   

             provincia  
86              CHUBUT  
1071  TIERRA DEL FUEGO  
15 

In [41]:
# Vemos un ejemplo
df_alquileres[df_alquileres['longitude'] == -65.029470]

id_alojamiento                                               name  \
70         20364091                                Departamento Solmar   
257        53295849  Preciosa unidad de alquiler de 3 dormitorios e...   

        category      rating           city  latitude  longitude  \
70   entire_home  4,82 (137)  Puerto Madryn -42.76963  -65.02947   
257  entire_home   4,84 (25)  Puerto Madryn -42.77073  -65.02947   

                                       characteristics    check_in  \
70   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-10-15   
257  ['3 camas', '2 dormitorios', '2 baños', '1 cam...  2023-07-03   

      check_out price_discounted price_original price_qualifier  \
70   2023-10-20              NaN        $47 USD           noche   
257  2023-07-08              NaN       $114 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
70              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
257             1000      2023-06-24       2023-05-01  239.919355    CHUBUT

*ACCION:*
<br>
*Entendemos que la columna longitude, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columan 8: characteristics

In [42]:
# Verificamos los duplicados
df_alquileres['characteristics'].duplicated(keep="first").sum()

243

In [43]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['characteristics'].duplicated(
    keep="first")].sample(10)

id_alojamiento                                              name  \
631            48806077                            Casa Pajaro Carpintero   
589            49107072                                   Contenedor-loft   
661            33512769                    Las Liebres Doble - D.T. 48/19   
625            53320962               Luminoso Departamento en Bariloche.   
439            51735487   Departamento con balcón en complejo residencial   
444  678351386809115205                Hermoso Departamento en Cipolletti   
474  829574033573661357                      Villa María, don Bosco 1125.   
161            31081731                Domos Puelo, para dos, con jacuzzi   
196            54214936          TINY LOFT 36 m2 NUEVO (centro de Esquel)   
401  719485899637095838  Departamento amueblado moderno para dos personas   

        category     rating                     city  latitude  longitude  \
631  entire_home  4,92 (37)  San Carlos de Bariloche -41.07006  -71.48875   
589  entire_home  4,91 (55)  San Carlos de Bariloche -41.13495  -71.32571   
661  entire_home  4,92 (90)  San Carlos de Bariloche -41.12717  -71.35206   
625  entire_home  4,97 (72)  San Carlos de Bariloche -41.13697  -71.29910   
439  entire_home        NaN               Cipolletti -38.92815  -67.99934   
444  entire_home        NaN               Cipolletti -38.93657  -67.99245   
474  entire_home        NaN                  Neuquén -38.96776  -68.06426   
161  entire_home  4,98 (40)               Lago Puelo -42.04157  -71.59237   
196  entire_home  4,72 (43)                   Esquel -42.91289  -71.31411   
401  entire_home        NaN                  Neuquén -38.97383  -68.05929   

                                       characteristics    check_in  \
631  ['3 camas', '2 dormitorios', '1 baño', '1 cama...  2023-07-01   
589  ['1 cama', '1 dormitorio', '1 baño', '1 cama k...  2023-10-26   
661  ['1 cama', '1 dormitorio', '1 baño', 'Estacion...  2023-09-26   
625  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-09-29   
439       ['1 cama', '1 dormitorio', '1 baño', 'Wifi']  2023-07-01   
444      ['2 camas', '1 dormitorio', '1 baño', 'Wifi']  2023-12-29   
474  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-07-01   
161  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-08   
196  ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-07-01   
401  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
631  2023-07-06          $90 USD       $100 USD           noche   
589  2023-11-01              NaN        $50 USD           noche   
661  2023-10-01              NaN        $50 USD           noche   
625  2023-10-04              NaN        $42 USD           noche   
439  2023-07-06              NaN        $51 USD           noche   
444  2024-01-03              NaN        $40 USD           noche   
474  2023-07-06              NaN        $55 USD           noche   
161  2023-07-15              NaN        $67 USD           noche   
196  2023-07-06              NaN        $52 USD           noche   
401  2023-07-06              NaN        $60 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
631             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
589             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
661             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
625             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
439              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
444              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
474              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
161             1000      2023-06-24       2023-05-01  239.919355     CHUBUT  
196             1000      2023-06-24       2023-05-01  239.919355     CHUBUT  
401           

*ACCION:*
<br>
*Entendemos que la columna characteristics, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 9: check_in

In [44]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['check_in'].duplicated(keep="first").sum()

920

*ACCION:*
<br>
*Entendemos que la columna check_in, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 10: check_out

In [45]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['check_out'].duplicated(keep="first").sum()

916

*ACCION:*
<br>
*Entendemos que la columna check_out, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 11: price_discounted

In [46]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_discounted'].duplicated(keep="first").sum()

938

*ACCION:*
<br>
*Entendemos que la columna price_discounted, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 12: price_original

In [47]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_original'].duplicated(keep="first").sum()

868

*ACCION:*
<br>
*Entendemos que la columna price_original, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 13: price_qualifier

In [48]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_qualifier'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna price_qualifier, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 14: cant_resultados

In [49]:
# Verificamos los duplicados
df_alquileres['cant_resultados'].duplicated(keep="first").sum()

1026

*ACCION:*
<br>
*Entendemos que la columna cant_resultados, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

Columna 15: fecha_scrapping

In [50]:
# Verificamos los duplicados
df_alquileres['fecha_scrapping'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna fecha_scrapping, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 16: fecha_cambio_bna

In [51]:
# Verificamos los duplicados
df_alquileres['fecha_cambio_bna'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna fecha_cambio_bna, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 17: cambio_bna

In [52]:
# Verificamos los duplicados
df_alquileres['cambio_bna'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna cambio_bna, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 18: provincia

In [53]:
# Verificamos los duplicados
df_alquileres['provincia'].duplicated(keep="first").sum()

1024

*ACCION:*
<br>
*Entendemos que la columna provincia, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

*OBSERVACION GENERAL*
<br>
*En general entendemos que la mayoria de las columnas podrian tener valores repetidos evaluandolas individualmente.*
<br>
*Pero si consideramos que la variable id_alojamiento identifica de manera biunivoca un alojamiento.*
<br>
*Podemos decir que no es lo mismo, un duplicado en la columna id_alojamiento(que no deberia tener valores repetidos), que un duplicado en la columna price_original,*
<br>
*(que podria perfectamente tener valores repetidos en la seria).*


**<h1>3. Análisis de datos faltantes y limpieza</h1>**

+ Mostrar el porcentaje de datos faltantes de cada columna

In [54]:
# Verificacion inicial del dataset
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1028 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_alojamiento    1028 non-null   int64  
 1   name              1028 non-null   object 
 2   category          1028 non-null   object 
 3   rating            842 non-null    object 
 4   city              1024 non-null   object 
 5   latitude          1028 non-null   float64
 6   longitude         1028 non-null   float64
 7   characteristics   1028 non-null   object 
 8   check_in          1028 non-null   object 
 9   check_out         1028 non-null   object 
 10  price_discounted  191 non-null    object 
 11  price_original    1028 non-null   object 
 12  price_qualifier   1028 non-null   object 
 13  cant_resultados   1028 non-null   int64  
 14  fecha_scrapping   1028 non-null   object 
 15  fecha_cambio_bna  1028 non-null   object 
 16  cambio_bna        1028 non-null   float64
 17  

*En primera instancia observamos que solo nos quedan dos columnas con datos faltantes: rating y price_discounted*
<br>
*Para rating, tenenmos un 81.90 % de faltantes (842 de 1028).*
<br>
*Para rating, tenenmos un 18.58 % de faltantes (191 de 1028).*


+ ¿Cuáles son las 3 columnas con más datos faltantes?

*En nuestra observacion inicial, en principio tenemos dos columnas con datos faltantes solamente: rating y price_discounted*

**LIMPIEZA DE DE DATOS POR COLUMNAS**

In [55]:
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1028 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_alojamiento    1028 non-null   int64  
 1   name              1028 non-null   object 
 2   category          1028 non-null   object 
 3   rating            842 non-null    object 
 4   city              1024 non-null   object 
 5   latitude          1028 non-null   float64
 6   longitude         1028 non-null   float64
 7   characteristics   1028 non-null   object 
 8   check_in          1028 non-null   object 
 9   check_out         1028 non-null   object 
 10  price_discounted  191 non-null    object 
 11  price_original    1028 non-null   object 
 12  price_qualifier   1028 non-null   object 
 13  cant_resultados   1028 non-null   int64  
 14  fecha_scrapping   1028 non-null   object 
 15  fecha_cambio_bna  1028 non-null   object 
 16  cambio_bna        1028 non-null   float64
 17  

+ Columna 0: id_alojamiento

In [56]:
df_alquileres['id_alojamiento'].nunique()

1028

In [57]:
df_alquileres['id_alojamiento'].isna().sum()

0

+ Columna 1: name

In [58]:
sorted(df_alquileres['name'].unique())

['"Alegria" vistas de ensueño en un lugar para enamorarse',
 '"Bathroom with LakeView Place and Retreat"',
 '"Casita" a estrenar, 21 m2 de luz',
 '"Cerros Nevados" departamento tipo loft.',
 '"Cipresales"     Cabaña  Villa la Angostura  Nqn',
 '"Doña Cota" - Amueblados Comahue',
 '"El Rancho" Sentirte en casa',
 '"La Encantada" en Villa los Coihues',
 '"La Posadita" Casa de Montaña, Miralejos',
 '"La casa de Cielo "',
 '"Los Amancays "\nCabaña de montaña',
 '"Tiny House de Ensueño"',
 '"Ty Bychan" \nLa Casita. Entre raíces galesas.',
 '0 - Pura Tranquilidad en el Corazón de Neuquén',
 '1- Pura Tranquilidad en el Corazón de Neuquén',
 '11-E Espectacular departamento Alto Neuquen',
 '2 - Pura tranquilidad en el corazón de Neuquén',
 '2 Amigos de viaje en un lugar comodo',
 '2 Habitaciones. 4 Personas en Casa con Pileta',
 '2 Personas. Lugar súper cómodo',
 '2 ambientes en el centro',
 '2x1 Central Family',
 '3 dormitorios. cama matrimonial + 3 de una plaza',
 '3- Pura tranquilidad en el 

+ Columna 2: category

In [59]:
# Vemos que tiene valores bien definidos, que no necesitan normalizar
df_alquileres['category'].unique()

array(['entire_home', 'private_room', 'shared_room'], dtype=object)

In [60]:
# Verificamos que no tienen  na
df_alquileres['category'].isna().sum()

0

In [61]:
# Cambiamos el tipo de la serie
df_alquileres['category'] = df_alquileres['category'].astype('category')

+ Columna 3: rating

In [62]:
# Vemos que tiene valores bien definidos, que no necesitan normalizar
df_alquileres['rating'].unique()

array([nan, '5,0 (6)', '4,91 (70)', 'Nuevo', '4,96 (95)', '4,86 (7)',
       '4,82 (17)', '4,96 (75)', '4,91 (67)', '4,57 (14)', '4,6 (5)',
       '4,94 (18)', '4,95 (22)', '4,92 (64)', '4,94 (109)', '4,77 (44)',
       '4,89 (35)', '5,0 (7)', '4,66 (53)', '5,0 (34)', '4,98 (93)',
       '4,96 (51)', '4,82 (66)', '4,8 (51)', '4,83 (36)', '4,9 (20)',
       '5,0 (17)', '4,65 (49)', '4,7 (33)', '4,89 (9)', '4,97 (90)',
       '5,0 (10)', '4,79 (30)', '4,86 (74)', '4,86 (44)', '4,63 (128)',
       '5,0 (3)', '4,88 (16)', '4,81 (32)', '5,0 (9)', '4,93 (27)',
       '4,92 (13)', '4,5 (4)', '4,77 (13)', '4,67 (12)', '4,9 (176)',
       '4,95 (19)', '4,8 (15)', '5,0 (14)', '4,79 (14)', '5,0 (4)',
       '5,0 (19)', '4,67 (6)', '4,82 (137)', '5,0 (11)', '4,77 (102)',
       '4,47 (55)', '4,76 (134)', '4,74 (19)', '4,96 (27)', '4,91 (11)',
       '4,93 (15)', '4,62 (13)', '4,55 (33)', '4,4 (5)', '4,72 (25)',
       '4,58 (76)', '4,73 (11)', '4,83 (60)', '4,93 (28)', '4,7 (10)',
       '4,76 (85

In [63]:
# Tratamiento de la columna rating
# Separamos los valores correspondientes para limpiarlos
df_rating= df_alquileres['rating'].str.split(pat="[ ]",expand=True).add_prefix('Columna')

# Renombramos las columnas
df_rating.rename(columns={"Columna0": "rating", "Columna1": "votaciones"}, inplace=True)

In [64]:
# Limpiamos los valores para la columna rating
df_rating['rating'] = df_rating['rating'].replace(',','.')

In [65]:
#  Rellenamos los valores para las columnas rating y votaciones
# Rellenamos con un valor promedio, pero establecemos que no tiene votaciones
df_rating.fillna({'rating':'3.00','votaciones':'0'},inplace= True)
# Quedo en la columna rating, con valor particular que debemos sustituir: Nuevo
df_rating['rating'] = np.where(df_rating['rating'].str.lower().str.contains('nuevo'), '3.00', df_rating['rating'])

In [66]:
# Reemplazamos la columna por punto para la columna rating y lo cambiamos de tipo
df_rating['rating'] = df_rating['rating'].str.replace(',','.')
df_rating['rating'] = df_rating['rating'].astype('float64')

In [67]:
# Limpiamos la columna de votaciones para cambiarle el tipo
df_rating['votaciones'] = df_rating['votaciones'].str.replace('(','').str.replace(')','')
df_rating['votaciones'] = df_rating['votaciones'].astype('int')

In [68]:
# Enriquecemos el dataframe de alquileres con las columnas de rating y votaciones
df_alquileres= df_alquileres.merge(df_rating,how='left',left_index=True,right_index=True,suffixes=('', '_y'))
df_alquileres[['rating_y','votaciones']].head()


rating_y  votaciones
0      3.00           0
1      3.00           0
2      5.00           6
3      4.91          70
4      3.00           0

In [69]:
# Actualizamos las columnas del dataframe rating y votaciones
df_alquileres['rating']= df_alquileres['rating_y']
df_alquileres = df_alquileres.drop('rating_y',axis=1)
df_alquileres.info()


<class 'pandas.core.frame.DataFrame'>
Index: 1028 entries, 0 to 1079
Data columns (total 19 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   id_alojamiento    1028 non-null   int64   
 1   name              1028 non-null   object  
 2   category          1028 non-null   category
 3   rating            1028 non-null   float64 
 4   city              1024 non-null   object  
 5   latitude          1028 non-null   float64 
 6   longitude         1028 non-null   float64 
 7   characteristics   1028 non-null   object  
 8   check_in          1028 non-null   object  
 9   check_out         1028 non-null   object  
 10  price_discounted  191 non-null    object  
 11  price_original    1028 non-null   object  
 12  price_qualifier   1028 non-null   object  
 13  cant_resultados   1028 non-null   int64   
 14  fecha_scrapping   1028 non-null   object  
 15  fecha_cambio_bna  1028 non-null   object  
 16  cambio_bna        1028 non-nu

+ Columna 4: city

In [90]:
# Vemos que tiene valores bien definidos, que no necesitan normalizar
s = pd.Series(df_alquileres['city'].unique())
s.sort_values()


70                                      AIU
67                                      AJK
66                                      AWF
41                                BARILOCHE
49                               BARILOCHE 
                      ...                  
46                      VILLA LA ANGOSTURA 
53    VILLA LA ANGOSTURA, NEUQUEN ARGENTINA
58                          VILLA LLAO LLAO
38                             VILLA TRAFUL
29                                      NaN
Length: 73, dtype: object

In [92]:
# Como vemos que varias ciudades son la misma pero con alguna diferencia tipografica, tratamos de normalizar los valores
df_alquileres['city'] = df_alquileres['city'].str.upper().trim
df_alquileres.head()

AttributeError: 'Series' object has no attribute 'trim'

In [75]:
df_alquileres['city'] = df_alquileres['city'].str.normalize('NFKD').str.encode('ascii', errors='ignore').str.decode('utf-8')
# df_alquileres['city'] = df_alquileres['city'].apply(remove_accents)
df_alquileres.head()

id_alojamiento                                               name  \
0  855050336912355203                                        North Beach   
1            51484364     Hermoso Monoambiente con Vista al Mar y Pileta   
2  800749301724167838                           Departamento Bella Vista   
3            50769917  New Lil’ & Cozy - 100mts de la playa Puerto Ma...   
4  916738743930751566                                        Front Beach   

      category  rating           city   latitude  longitude  \
0  entire_home    3.00  PUERTO MADRYN -42.756301 -65.037517   
1  entire_home    3.00  PUERTO MADRYN -42.755775 -65.037265   
2  entire_home    5.00  PUERTO MADRYN -42.768007 -65.033108   
3  entire_home    4.91  PUERTO MADRYN -42.777590 -65.025890   
4  entire_home    3.00  PUERTO MADRYN -42.770540 -65.030810   

                                     characteristics    check_in   check_out  \
0              ['2 camas', '1 dormitorio', '1 baño']  2023-07-06  2023-07-12   
1  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-21  2023-07-26   
2  ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-08-01  2023-08-08   
3  ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-09  2023-07-14   
4              ['2 camas', 'Monoambiente', '1 baño']  2023-07-01  2023-07-06   

  price_discounted price_original price_qualifier  cant_resultados  \
0              NaN        $38 USD           noche             1000   
1          $18 USD        $24 USD           noche             1000   
2              NaN        $43 USD           noche             1000   
3              NaN        $41 USD           noche             1000   
4              NaN        $35 USD           noche             1000   

  fecha_scrapping fecha_cambio_bna  cambio_bna provincia  votaciones  
0      2023-06-24       2023-05-01  239.919355    CHUBUT           0  
1      2023-06-24       2023-05-01  239.919355    CHUBUT           0  
2      2023-06-24       2023-05-01  239.919355    CHUBUT           6  
3      2023-06-24       2023-05-01  239.919355    CHUBUT          70  
4      2023-06-24       2023-05-01  239.919355    CHUBUT           0